tests/flow/test_comprehension_functions.py

```diff
@@ -462,3 +464,24 @@ def test20_pattern_comprehension_in_switch_case(self):
         expected_result = [[[1, 1]], [[2]], [[3]], [[]]]
         self.env.assertEquals(actual_result.result_set, expected_result)
 
+    def test21_pattern_comprehension_scope(self):
+        # variables included by the pattern comprehension can't be available
+        # out of its scope
+        queries = [
+            "MATCH (a) RETURN [p = (a)-[e]->(f) WHERE e.v < 7 | f] as z, p",
+            "MATCH (a) RETURN [p = (a)-[e]->(f) WHERE e.v < 7 | f] as z, e",
+            "MATCH (a) RETURN [p = (a)-[e]->(f) WHERE e.v < 7 | f] as z, f",
+        ]
+        for query in queries:
+            self._assert_exception(redis_graph, query, "not defined")
+
+    def test22_invalid_pattern_comprehension(self):
+        # test error detection of invalid pattern comprehension
+        queries_with_errors = [
+            ("MATCH (a) RETURN [(b)-[b]->() | 0]", "The alias 'b' is already defined"),
```

Comment:
currently the validation for this query is 
The alias 'b' was specified for both a node and a relationship
which is better try to find out if we can back to it